docs/index.md

```diff
@@ -0,0 +1,279 @@
+
+##### Set variables
+
+To expedite copy/paste without the needing to update placeholders manually, update and execute the following.
+
+```bash
+export CREDENTIALS_BLOCK_NAME="aws-credentials"
+export VPC_ID="vpc-id"
+export ECS_TASK_BLOCK_NAME="ecs-task-example"
+export S3_BUCKET_BLOCK_NAME="ecs-task-bucket-example"
+```
+
+##### Save an infrastructure and storage block
+
+Save a custom infrastructure and storage block by executing the following snippet.
+
+```python
+import os
+from prefect_aws import AwsCredentials, ECSTask, S3Bucket
+
+aws_credentials = AwsCredentials.load(os.environ["CREDENTIALS_BLOCK_NAME"])
+
+ecs_task = ECSTask(
+    image="prefecthq/prefect:2-python3.10",
+    aws_credentials=aws_credentials,
+    vpc_id=os.environ["VPC_ID"],
+)
+ecs_task.save(os.environ["ECS_TASK_BLOCK_NAME"], overwrite=True)
+
+bucket_name = "ecs-task-bucket-example"
+s3_client = aws_credentials.get_s3_client()
+s3_client.create_bucket(
+    Bucket=bucket_name,
+    CreateBucketConfiguration={"LocationConstraint": aws_credentials.region_name}
+)
+s3_bucket = S3Bucket(
+    bucket_name=bucket_name,
+    aws_credentials=aws_credentials,
+)
+s3_bucket.save(os.environ["S3_BUCKET_BLOCK_NAME"], overwrite=True)
+```
+
+##### Write a flow
+
+Then, use an existing flow to create a deployment with, or use the flow below if you don't have an existing flow handy.
+
+```python
+from prefect import flow
+
+@flow(log_prints=True)
+def ecs_task_flow():
+    print("Hello, Prefect!")
+
+if __name__ == "__main__":
+    ecs_task_flow()
+```
+
+##### Create a deployment
+
+If the script was named "ecs_task_script.py", build a deployment manifest with the following command.
+
+```bash
+prefect deployment build ecs_task_script.py:ecs_task_flow \
+    -n ecs-task-deployment \
+    -ib ecs-task/${ECS_TASK_BLOCK_NAME} \
+    -sb s3-bucket/${GCS_BUCKET_BLOCK_NAME} \
+    --override env.EXTRA_PIP_PACKAGES=prefect-aws
+```
+
+Now apply the deployment!
+
+```bash
+prefect deployment apply ecs_task_flow-deployment.yaml
+```
+
+##### Test the deployment
+
+Start up an agent in a separate terminal. The agent will poll the Prefect API for scheduled flow runs that are ready to run.
+
+```bash
+prefect agent start -q 'default'
+```
+
+Run the deployment once to test.
+
+```bash
+prefect deployment run ecs-task-flow/ecs-task-deployment
+```
+
+Once the flow run has completed, you will see `Hello, Prefect!` logged in the Prefect UI.
+
+!!! info "No class found for dispatch key"
+
+    If you encounter an error message like `KeyError: "No class found for dispatch key 'ecs-task' in registry for type 'Block'."`,
+    ensure `prefect-aws` is installed in the environment that your agent is running!
+
+Another tutorial on `ECSTask` can be found [here](https://towardsdatascience.com/prefect-aws-ecs-fargate-github-actions-make-serverless-dataflows-as-easy-as-py-f6025335effc).
+
+#### Within Flow
+
+You can execute commands through ECS Task directly within a Prefect flow.
```

Comment:
I think it's useful to include why this would be something you'd want to do.
```suggestion
You can also execute commands through ECS Task directly within a Prefect flow. Running containers via ECS in your flows is useful for executing non-python code in a distributed manner via Prefect.
```